src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -915,6 +864,14 @@ ModifyQuerySupported(Query *queryTree, Query *originalQuery, bool multiShardQuer
 	{
 		ExtractRangeTableEntryWalker((Node *) originalQuery, &rangeTableList);
 	}
+	RangeTblEntry *resultRte = ExtractResultRelationRTE(queryTree);
+	Oid resultRelationId = InvalidOid;
+	if (resultRte)
+	{
+		resultRelationId = resultRte->relid;
+	}
+	bool containsTableToBeConvertedToSubquery =
```

Comment:
here if we only use `ContainsLocalTableDistributedTableJoin` then it would return true for a reference-citus_local table join. However if the data is already locally available we don't wanna give an error, as in the router planner should be able to handle it right? I am a bit surprised that router planner doesn't already understand that data is locally available Am I missing something here?

Reply:
See https://github.com/citusdata/citus/pull/4358/commits/ecfd775eb7949a078a88157a4e7ea09db3b6e808